pkg/dockerregistry/server/manifestservice.go

```diff
@@ -174,9 +174,9 @@ func (m *manifestService) Put(ctx context.Context, manifest distribution.Manifes
 		}
 
 		status := statusErr.ErrStatus
-		if status.Code != http.StatusNotFound ||
-			(strings.ToLower(status.Details.Kind) != "imagestream" /*pre-1.2*/ && strings.ToLower(status.Details.Kind) != "imagestreams") ||
-			status.Details.Name != m.repo.name {
+		kind := strings.ToLower(status.Details.Kind)
+		isValidKind := kind == "imagestream" /*pre-1.2*/ || kind == "imagestreams" /*1.2 to 1.6*/ || kind == "imagestreammappings" /*1.7+*/
+		if !isValidKind || status.Code != http.StatusNotFound || status.Details.Name != m.repo.name {
```

Comment:
@legionus @mfojtik PTAL

This change is required because in `pkg/image/registry/imagestreammapping/rest.go`:

```go
// findStreamForMapping retrieves an ImageStream whose DockerImageRepository matches dockerRepo.
func (s *REST) findStreamForMapping(ctx apirequest.Context, mapping *imageapi.ImageStreamMapping) (*imageapi.ImageStream, error) {
	if len(mapping.Name) > 0 {
		// Since ctx is passed through here, this returns a NotFound error for imagestreammappings now
		return s.imageStreamRegistry.GetImageStream(ctx, mapping.Name, &metav1.GetOptions{})
	}
...
}
```